contracts/Pair.sol

```diff
@@ -1,129 +1,164 @@
 
-    uint public reserve0CumulativeLast;
-    uint public reserve1CumulativeLast;
+    uint256 public reserve0CumulativeLast;
+    uint256 public reserve1CumulativeLast;
 
     // index0 and index1 are used to accumulate fees, this is split out from normal trades to keep the swap "clean"
     // this further allows LP holders to easily claim fees for tokens they have/staked
-    uint public index0 = 0;
-    uint public index1 = 0;
+    uint256 public index0 = 0;
+    uint256 public index1 = 0;
 
     // position assigned to each LP to track their current index0 & index1 vs the global position
-    mapping(address => uint) public supplyIndex0;
-    mapping(address => uint) public supplyIndex1;
+    mapping(address => uint256) public supplyIndex0;
+    mapping(address => uint256) public supplyIndex1;
 
     // tracks the amount of unclaimed, but claimable tokens off of fees for token0 and token1
-    mapping(address => uint) public claimable0;
-    mapping(address => uint) public claimable1;
-
-    event Fees(address indexed sender, uint amount0, uint amount1);
-    event Mint(address indexed sender, uint amount0, uint amount1);
-    event Burn(address indexed sender, uint amount0, uint amount1, address indexed to);
+    mapping(address => uint256) public claimable0;
+    mapping(address => uint256) public claimable1;
+
+    event Fees(address indexed sender, uint256 amount0, uint256 amount1);
+    event TankFees(address indexed token, uint256 amount0, address tank);
+    event GaugeFees(address indexed token, uint256 amount0, address externalBribe);
+    event Mint(address indexed sender, uint256 amount0, uint256 amount1);
+    event Burn(address indexed sender, uint256 amount0, uint256 amount1, address indexed to);
     event Swap(
         address indexed sender,
-        uint amount0In,
-        uint amount1In,
-        uint amount0Out,
-        uint amount1Out,
+        uint256 amount0In,
+        uint256 amount1In,
+        uint256 amount0Out,
+        uint256 amount1Out,
         address indexed to
     );
-    event Sync(uint reserve0, uint reserve1);
-    event Claim(address indexed sender, address indexed recipient, uint amount0, uint amount1);
+    event Sync(uint256 reserve0, uint256 reserve1);
+    event Claim(address indexed sender, address indexed recipient, uint256 amount0, uint256 amount1);
 
-    event Transfer(address indexed from, address indexed to, uint amount);
-    event Approval(address indexed owner, address indexed spender, uint amount);
+    event Transfer(address indexed from, address indexed to, uint256 amount);
+    event Approval(address indexed owner, address indexed spender, uint256 amount);
 
     constructor() {
         factory = msg.sender;
-        (address _token0, address _token1, bool _stable) = PairFactory(msg.sender).getInitializable();
+        //voter = PairFactory(msg.sender).voter(); // nice easy way to add the voter :) we already getting this from pair factory tho
```

Comment:
why don't we set voter here and use it to guard `setHasGauge` and `setExternalBribe`?